grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -177,6 +179,47 @@ class StockMovementService {
         }
     }
 
+    void validateQuantityRequested(StockMovement stockMovement) {
+        stockMovement.lineItems?.each { StockMovementItem item ->
+            if (item.substitutionItems) {
+                item.substitutionItems.each {
+                    if (!validateQuantityRequested(it)) {
+                        stockMovement.errors.reject("stockMovement.invalidQtyRequested.message", [it.product?.name] as Object[], "Invalid quantity requested for product " + it.product?.name)
+                    }
+                }
+            } else {
+                if (!validateQuantityRequested(item)) {
+                    stockMovement.errors.reject("stockMovement.invalidQtyRequested.message", [item.product?.name] as Object[], "Invalid quantity requested for product " + item.product?.name)
+                }
+            }
+        }
+
+        if (!stockMovement.validate()) {
+            throw new ValidationException("Invalid stock movement", stockMovement.errors)
+        }
+    }
+
+    Boolean validateQuantityRequested(StockMovementItem item) {
+        RequisitionItem requisitionItem = item.requisitionItem
+        Location location = requisitionItem?.requisition?.origin
+        Integer quantityRequired = requisitionItem?.calculateQuantityRequired()
+
+        if (requisitionItem.modificationItem) {
+            requisitionItem = requisitionItem.modificationItem
+        }
+
+        def quantityAvailable = inventoryService.getQuantityAvailableToPromise(item.product, location)
+        if (requisitionItem.picklistItems) {
+            def qtyPicked = requisitionItem.picklistItems.findAll { it.pickable }?.sum { it.quantity }
+            quantityAvailable += (qtyPicked ?: 0)
```

Comment:
It look me a few mintues to understand what was happening here. My takeaway is that you're adding the amount picked to the quantity available because otherwise this would fail validation. Could you add a comment above line 211 just to describe what's happening from your perspective.